gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -0,0 +1,195 @@
+import org.hl7.v3.RCMRMT030101UK04Component2;
+import org.hl7.v3.RCMRMT030101UK04Component3;
+import org.hl7.v3.RCMRMT030101UK04Component4;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04EhrExtract;
+import org.hl7.v3.RCMRMT030101UK04EhrFolder;
+import org.hl7.v3.RCMRMT030101UK04MedicationStatement;
+import org.hl7.v3.RCMRMT030101UK04Prescribe;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+
+@Service
+@AllArgsConstructor
+public class MedicationRequestMapper {
+    private MedicationMapper medicationMapper;
+    private MedicationRequestOrderMapper medicationRequestOrderMapper;
+    private MedicationRequestPlanMapper medicationRequestPlanMapper;
+    private MedicationStatementMapper medicationStatementMapper;
+
+    public List<DomainResource> mapResources(RCMRMT030101UK04EhrExtract ehrExtract, List<Encounter> encounters, Patient patient) {
+
+        List<RCMRMT030101UK04EhrComposition> ehrCompositions = ehrExtract.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component::getEhrFolder)
+            .map(RCMRMT030101UK04EhrFolder::getComponent)
+            .flatMap(List::stream)
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .collect(Collectors.toList());
+
+        for (RCMRMT030101UK04EhrComposition ehrComposition : ehrCompositions) {
+            var context = encounters.stream()
+                .filter(encounter1 -> encounter1.getId().equals(ehrComposition.getId().getRoot())).findFirst();
+
+            return ehrComposition.getComponent()
+                .stream()
+                .filter(RCMRMT030101UK04Component4::hasMedicationStatement)
+                .map(RCMRMT030101UK04Component4::getMedicationStatement)
+                .map(medicationStatement -> mapMedicationStatement(ehrExtract, ehrComposition, medicationStatement, patient, context))
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+        }
+        return List.of();
+    }
+
+    private List<DomainResource> mapMedicationStatement(RCMRMT030101UK04EhrExtract ehrExtract,
+        RCMRMT030101UK04EhrComposition ehrComposition, RCMRMT030101UK04MedicationStatement medicationStatement,
+        Patient subject, Optional<Encounter> context) {
+
+        var authoredOn = extractAuthoredOn(ehrComposition,
+            DateFormatUtil.parseToDateTimeType(ehrExtract.getAvailabilityTime().getValue()));
+        var requester = extractRequester(ehrComposition, medicationStatement);
+        var recorder = extractRecorder(ehrComposition, medicationStatement);
+
+        List<Medication> medications = medicationStatement.getConsumable()
+            .stream()
+            .map(medicationMapper::createMedication)
+            .collect(Collectors.toList());
+
+        List<MedicationRequest> medicationRequestsOrder = medicationStatement.getComponent()
```

Comment:
I'd consider creating separate methods for all of those streams here that return List<MedicationRequest>